### ISSUE TITLE
Criterion has a confusing signature that relies on coupled behavior from the model's return.

### ISSUE THREAD

**Issue by merterden98** [Yuanhaojun513/MMUNet#3]
https://github.com/Yuanhaojun513/MMUNet/blob/964d4e328b0ae02d2d9f62dbfdbe4d7ec1b778a6/train_utils/train_and_eval.py#L8

Update the criterion to use `logits` directly instead of iterating.
